src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1512,6 +1512,25 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnSrpClientNotificatio
     case OT_ERROR_NONE: {
         ChipLogProgress(DeviceLayer, "OnSrpClientNotification: Last requested operation completed successfully");
 
+        if (aHostInfo)
+        {
+            if (aHostInfo->mState == OT_SRP_CLIENT_ITEM_STATE_REMOVED)
+            {
+                // Clear memory for removed host
+                memset(ThreadStackMgrImpl().mSrpClient.mHostName, 0, sizeof(ThreadStackMgrImpl().mSrpClient.mHostName));
+
+                if (!ThreadStackMgrImpl().mSrpClient.mInitializedCallback)
```

Comment:
Is this even possible?

Reply:
Considering overall current Matter implementation - I think not. I added it, as I've got bad experience if it comes to the mDNS and SRP layers changes, so I can imagine regression that may appear if methods call order will change, so someone will not set callbacks before requesting publishing service and we would fall into nullptr call here. If we had unit tests for SRP I would sleep better ...

Reply:
I changed my mind. I also added nullptr check in CleanSrpHost(), so I think this is useless, removing it.